### ISSUE TITLE
Neovim freezes on viewing lua files, goes back to normal on highlight = false [MacOS Ventura 13.4]

### ISSUE THREAD

**Issue by rootex-** [nvim-treesitter/nvim-treesitter#5294]
### Describe the bug

I open any lua file and nvim starts to freeze, becomes unresponsive.

If I disable nvim-treesitter highlighting: 
```lua
      highlight = {
        -- `false` will disable the whole extension
        enable = false,
```
the problem disappears. 

### To Reproduce

1. Open any lua file.
2. Nvim becomes unresponsive, freezes, very long delays between any actions.
3. Quit nvim.
4. Set nvim-treesitter highlight to false.
5. Run nvim. 
6. Open any lua file. 
7. Nvim works as expected, no freezes or long delays.

### Expected behavior
No freezing or very long delays between user actions.

### Output of `:checkhealth nvim-treesitter`

```markdown
==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.20.8 (0c49d6745b3fc4822ab02e0018770cd6383a779c) (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v20.5.1 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: Apple clang version 14.0.3 (clang-1403.0.22.14.1)
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "22.5.0",
  sysname = "Darwin",
  version = "Darwin Kernel Version 22.5.0: Mon Apr 24 20:51:50 PDT 2023; root:xnu-8796.121.2~5/RELEASE_X86_64"
} ~

Parser/Features         H L F I J
  - ada                 ✓ ✓ ✓ . .
  - bash                ✓ ✓ ✓ . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - clojure             ✓ ✓ ✓ . ✓
  - comment             ✓ . . . .
  - commonlisp          ✓ ✓ ✓ . .
  - cpp                 ✓ ✓ ✓ ✓ ✓
  - css                 ✓ . ✓ ✓ ✓
  - diff                ✓ . . . .
  - dockerfile          ✓ . . . ✓
  - git_rebase          ✓ . . . ✓
  - gitattributes       ✓ . . . ✓
  - gitcommit           ✓ . . . ✓
  - gitignore           ✓ . . . .
  - go                  ✓ ✓ ✓ ✓ ✓
  - groovy              ✓ . . . ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - http                ✓ . . . ✓
  - java                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - jq                  ✓ . . . ✓
  - jsdoc               ✓ . . . .
  - json                ✓ ✓ ✓ ✓ .
  - kdl                 ✓ ✓ ✓ ✓ ✓
  - kotlin              ✓ ✓ ✓ . ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - org                 . . . . .
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - scala               ✓ ✓ ✓ . ✓
  - sql                 ✓ . . ✓ ✓
  - toml                ✓ ✓ ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓
  - yaml                ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~
```

### Output of `nvim --version`

```text
NVIM v0.9.1
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.9.1/share/nvim"

Run :checkhealth for more info
```

**Comment by lucario387** [comment]
It's the comment parser again, probably

Can you try uninstalling it and experience again

**Comment by uolot** [comment]
> If I disable nvim-treesitter highlighting:
> 
> ```lua
>       highlight = {
>         -- `false` will disable the whole extension
>         enable = false,
> ```
> 
> the problem disappears.

I experience the same issue. Instead of disabling it in the config, `:TSDisable hightlight` also fixes it in the current session.

@lucario387 Re-installing `lua` parser doesn't help.

**Comment by uolot** [comment]
@rootex- instead of disabling highlighting entirely, you can set this for now:

```lua
     highlight = {
        enable = true,
        disable = { "lua" },
```

**Comment by lucario387** [comment]
I mean the comment parser, not the lua parser.
Most of the issues we've had so far is related to the comment parser causing performance issues so uninstall it and try again

**Comment by lucario387** [comment]
Also, another related issue to it might be misinterpreted injections, which would be fixed in 0.9.2
